update misc/hoverline (#66)
OI-wiki/i18n: docs/misc/hoverline.md

Hover line is a vertical line. It has two properties: initial position and height. It can move left and right when its upper bound does not exceed the height of the rectangle at the current position.

For a hover line, we move it left or right under the premise that the upper bound does not exceed the height of the rectangle at the current position and does not move out of the boundary, and find out where it can expand to left and right at most. At this time, the area swept by this hover line is the largest possible rectangle containing it. It is easy to find that the largest sub-rectangle must contain a suspension line with an initial position of $i$ and a height of $h_i$. The time complexity of enumeration to achieve this process is $O(n ^ 2)$, but we can optimize it to $O(n)$ using the suspension method.

We consider how to quickly find the leftmost endpoint that the hover line can reach.

Define $l_i$ as the leftmost endpoint to which the hover line of the currentl $i$-th position can reach. It is easy to get the initial value of $l_i$ as $i$ . We need to further check whether it can expand to the left.

> - If the current $l_i = 1$ , then it has expanded to the boundary. Therefore, it cannot.

Reviewer: 建议 “不可以” 统一表述。
我的建议是第一行改成第二行的表述

It seems that the hover line can be replaced and is useless, but its concept is simpler than the monotonic stack, and is more suitable for OI beginners to understand and use to solve problems like the largest sub-rectangle.

> ???+note "[SP1805 HISTOGRA - Largest Rectangle in a Histogram](https://www.luogu.com.cn/problem/SP1805)"

Reviewer: ```suggestion
???+note "[SP1805 HISTOGRA - Largest Rectangle in a Histogram](https://www.spoj.com/problems/HISTOGRA/)"
```

    }
    ```

???+note "[UVA1619 Feel Good](https://www.luogu.com.cn/problem/UVA1619)"
    For a sequence of length $n$ , find a sub-interval so that the product of the minimum value in the sub-interval and the length of the sub-interval is the maximum. It is required to minimize the length while meeting the maximum comfort value, and to minimize the number of the left endpoint while the length is minimum.

> In this question, we can enumerate the minimum value. We consider the number $a_i$ at each position as the minimum value, expanding from $i$ to the left and right, and find that $\min\limits _ {j = l} ^ r a_j = a_i$ as much as possible. In this way, this question has been transformed into a hover line model.

Reviewer: as much as possible 🤔

Author: try our best? as much as we could? 🤔

Reviewer: 不是 感觉这样连一个 as much as possible 感觉有点不通顺，没说清楚是一个拓展的区间

    }
    ```

> ## Largest submatrix

Reviewer: 不是矩阵啊:new_moon_with_face:

???+note "[P4147 Yuchan Palace](https://www.luogu.com.cn/problem/P4147) (original link in Chinese)"

>     Given a $n \times m$ matrix containing `'F'` and `'R'`, find the area $\times 3$ of the largest submatrix so that the value of each element is `'F'`.

Reviewer: ```suggestion
    Given a $n \times m$ matrix consists of`'F'` and `'R'`, find the area $\times 3$ of the largest submatrix so that the value of all elements in this matrix are `'F'`.
```
